modules/core/src/smithy4s/Endpoint.scala

```diff
@@ -66,4 +66,103 @@ object Endpoint {
 
   }
 
+  type ForOperation[Op[_, _, _, _, _]] = {
+    type e[I, E, O, SI, SO] = Endpoint[Op, I, E, O, SI, SO]
+  }
+
+  final case class Builder[Op[_, _, _, _, _], I, E, O, SI, SO] private (
+      private val baseId: ShapeId,
+      private val baseHints: Hints,
+      private val baseInput: Schema[I],
+      private val baseOutput: Schema[O],
+      private val baseErrorable: Option[Errorable[E]],
+      private val baseStreamedInput: StreamingSchema[SI],
+      private val baseStreamedOutput: StreamingSchema[SO],
+      private val baseWrap: I => Op[I, E, O, SI, SO]
```

Comment:
**thought:** I didn't suggest adding the ability to customize `wrap` because, frankly, what can you do? other than side effects like printing the input, I don't see how else you might extend this. Just leaving this for other maintainers if they're wondering why `wrap` doesn't have builder functions.

Reply:
if customization of the wrap is not available, we must offer an alternative solution that involves providing an implementation. Its seems unavoidable in this case

Reply:
I had a thought the other day, now I'm thinking `wrap` could actually be customizable on an individual endpoint level. so, `withWrap` and `mapWrap` would still be doable.

I think we can leave them out of this PR though, and make a separate ticket.